lib/resty/core/shdict.lua

```diff
@@ -380,9 +380,31 @@ local function shdict_incr(zone, key, value, init)
         init = 0
     end
 
+    if exptime ~= nil then
+        local typ = type(exptime)
+        if typ ~= "number" then
+            exptime = tonumber(exptime)
+
+            if not exptime then
+                error("bad exptime arg: number expected, got " .. typ, 2)
+            end
+        end
+
+        if exptime < 0 then
+            error('bad "exptime" argument', 2)
+        end
+
+        if has_init == 0 then
```

Comment:
We should use a boolean typed value here instead of using an integer? It's Lua anyway. Seems like an old issue.

Reply:
Do you want me to update this logic in the above handling of the `init` argument in the same commit to use a boolean instead?

Reply:
@thibaultcha Yes, please.

Reply:
@agentzh Updated, got rid of the `has_init` variable altogether.